File: solutions/python/algorithm/set/Exercise3.py
from structure.dynamic.DynamicSetADT import DynamicSetADT


def filter_greater_than(original_set, threshold):
    result = DynamicSetADT()
    while not original_set.isEmpty():
        value = original_set.choose()
        if value > threshold:
            result.add(value)
        original_set.remove(value)
    return result


# Crear conjunto original
my_set = DynamicSetADT()
for val in [5, 3, 8, 1, 6]:
    my_set.add(val)

# Filtrar valores mayores a 4
filtered_set = filter_greater_than(my_set, 4)

print("Elementos mayores a 4:")
while not filtered_set.isEmpty():
    val = filtered_set.choose()
    print(val, end=" ")
    filtered_set.remove(val)

File: solutions/javascript/src/algorithm/set/Exercise3.js
import DynamicSetADT from "../../structure/dynamic/DynamicSetADT.js";

function filterGreaterThan(originalSet, threshold) {
  const result = new DynamicSetADT();

  while (!originalSet.isEmpty()) {
    const value = originalSet.choose();
    if (value > threshold) {
      result.add(value);
    }
    originalSet.remove(value);
  }

  return result;
}


const mySet = new DynamicSetADT();
[5, 3, 8, 1, 6].forEach(v => mySet.add(v));

const filteredSet = filterGreaterThan(mySet, 4);

console.log("Elementos mayores a 4:");
while (!filteredSet.isEmpty()) {
  const val = filteredSet.choose();
  console.log(val);
  filteredSet.remove(val);
}